src/WebUI/src/composables/item/use-item-upgrades.ts

```diff
@@ -0,0 +1,32 @@
+import { type ItemFlat } from '@/models/item';
+import { getItemUpgrades, getCompareItemsResult } from '@/services/item-service';
+import { createItemIndex } from '@/services/item-search-service/indexator';
+import { type AggregationConfig } from '@/models/item-search';
+
+export const useItemUpgrades = (
+  item: ItemFlat,
+  cols: AggregationConfig,
+  excludeBaseItem = false
+) => {
+  const { state } = useAsyncState(() => getItemUpgrades(item.baseId), []);
+
+  const itemUpgrades = computed((): ItemFlat[] => {
+    return (
+      createItemIndex(
+        excludeBaseItem
+          ? state.value.filter(itemUpgrade => itemUpgrade.id !== item.id)
+          : state.value,
+        false
+      )
+        // TODO: hotfix, avoid duplicate items with multiply weaponUsage
+        .filter((value, index, self) => index === self.findIndex(t => t.id === value.id))
```

Comment:
remove comment as it's been done?

Reply:
keep it for now.
As soon as I'm free, I'll improve and cover it with tests. Also, in the current implementation is not possible to see the stats of alternate weapon mode

Reply:
ref: https://github.com/verdie-g/crpg/issues/992